tests/unit/test_tasks.py

```diff
@@ -225,7 +226,18 @@ def test_cleanups_only_ran_once_on_exception(self):
         for done_callback in done_callbacks:
             self.transfer_coordinator.add_done_callback(done_callback)
 
-        submission_task()
+        try:
+            submission_task()
+        except InvalidStateTransitionError:
```

Comment:
I don't think we should be doing this.  The point of that exception is that seeing an invalid state transition is indicative of a bug as you mentioned in a previous comment.

Also, what's to prevent this from happening in the "real" code?  Seems like there's a chance a user could actually see an `InvalidStateTransitionError`